### ISSUE TITLE
[3.7.2] [Heavy Bug] Slope Map don't works and kills system connection to ext sdcard

### ISSUE THREAD

**Issue by Thunichgud** [osmandapp/Osmand#9066]
After downloading the German slope map (named in German: "Pisten" - very stupid) and after trying to switch from hillshades to slope map my device loses connection to the external sdcard, where all Osmand maps and data are stored. The hillshades disappears but the slope map never appears. No app knows from this moment the external sdcard and all playlists of my music are destroyed.
I have to reboot the device multiple times and after that I have to rebuild my playlists.
This happens every time, when I try to switch to the slope map.
Device: Samsung Galaxy S4
CPU: armv7
OS: CyanogenMod 12.1 (=Android 5.1)

**Comment by vshcherb** [comment]
Looks like SD-card got corrupted because of too much of new data. DIdn't experience it yet though it is quite typical cause saving lots of new data could cause corruption of old data if the SD card is old.

**Comment by sonora** [comment]
Yes, I concur. Sounds like a flaky SD card, or even a fake SD card (not having the capacity it claims to have), I've seen both phenomenons on different occasions before. A way to recover that is to confirm by testing a new SD card, then copy everything from old to new using a USB card reader.

**Comment by Thunichgud** [comment]
I tested your hints. I tested it with the old SD card which I formatted and tested with h2testw.exe v1..4 and I tested it with a new SD card which i also tested with h2testw.exe v1.4. Both SD cards are OK. On both SD cards I observed this issue sometimes and sometimes I don't observed it. I don't know when and why.
It seems that sometimes at the first activation of the slope file OSMand don't produces the correct belonging "Slope Germany europe. sqlitedb-shm" and "Slope Germany europe. sqlitedb-wal" files. Sometimes one of them is missing and sometimes one of them is empty (0.0 B). After that, the system crashes every time when starting OSMand with activated slope files. But sometimes OSMand produces at the first activation of the slope file the correct sqlitedb files and after that, OSMand works correct with activatet slope files. For the tests, I deleted the sqlitdb files as a simulation of first activation.

**Comment by Thunichgud** [comment]
I observed this issue at 3.7.3 and 3.7.4, too.

**Comment by vshcherb** [comment]
This is strange and we will keep monitoring it, though it is not clear yet why sqlite misbehaves sometimes
